source/Halibut/Transport/SecureListener.cs

```diff
@@ -262,12 +252,10 @@ async Task ExecuteRequest(TcpClient client)
                 }
                 finally
                 {
-                    if (!keepStreamOpen)
-                    {
-                        // Closing an already closed stream or client is safe, better not to leak
-                        stream.Close();
-                        client.Close();
-                    }
+                    // Closing an already closed stream or client is safe, better not to leak
+                    stream.Close();
+                    client.Close();
+                    tcpClientManager.RemoveDisconnectedClient(client);
```

Comment:
The `close` statements could throw an exception, so perhaps this should go first.

Reply:
Could `tcpClientManager.RemoveDisconnectedClient(client);` throw?

Reply:
@tothegills - dont think so...

Reply:
I looked at the `stream.Close()` methods a long time ago, and they appeared safe. However, after all the leak problems we've had, I would feel more confident if there was some super defensive code in here. Either @shane's suggestion, or `try/catch` each statement.